sdk/core/core-client-rest/src/apiVersionPolicy.ts

```diff
@@ -21,7 +21,7 @@ export function apiVersionPolicy(options: ClientOptions): PipelinePolicy {
       if (!url.searchParams.get("api-version") && options.apiVersion) {
         // append the apiVersion with client one
         url.searchParams.append("api-version", options.apiVersion);
-        req.url = url.toString();
+        req.url = `${req.url}&api-version=${options.apiVersion}`;
```

Comment:
In the apiVersionPolicy, we use url.toString() to get the final url, which will encode everything and ruin our effort in buildRequestUrl even if we have passed skipUrlEncoding.

Reply:
the above doesn't do any checks on if `url` has any existing query parameters. For example if `url` is `https://example.com/foo` this will create `https://example.com/foo&api-version=123` instead of `https://example.com/foo?api-version=123`